sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs

```diff
@@ -440,7 +440,7 @@ Task ErrorHandler(ProcessErrorEventArgs args)
                 }
                 await processor.StartProcessingAsync();
                 var stopwatch = Stopwatch.StartNew();
-                while (stopwatch.Elapsed.TotalSeconds <= 10)
+                while (stopwatch.Elapsed.TotalSeconds <= 30)
```

Comment:
This seems like it may have some non-determinism due to the timings.   You may want to consider setting a `TaskCompletionSource` in your exception handler and awaiting that here with a `Task.WaitAny` timeout.   I'd also advise allowing for a few minutes in that timeout due to how things run in CI.   _(Event Hubs had some issues with shorter timeouts like this when parallelism caused a test to suspend for longer than we expected)_

To illustrate, I'm thinking something like:

```csharp
using var cancellationSource = new CancellationTokenSource();
cancellationSource.CancelAfter(ServiceBusTestEnvironment.Instance.TestExecutionTimeLimit);

// stuff...

await Task.WhenAny(exceptionHandlerCalledSource.Task, Task.Delay(Timeout.Infinite, cancellationSource.Token));
Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The cancellation token should not have been signaled.");
cancellationSource.Cancel();

// stuff...
```

Reply:
That's fair - this was copied wholesale from Track 1. If there is any more flakiness I will make this update.